python/nano/src/bigdl/nano/pytorch/trainer/Trainer.py

```diff
@@ -216,24 +215,14 @@ def quantize(self, pl_model: LightningModule,
                                         tuning_strategy=tuning_strategy,
                                         accuracy_criterion=accuracy_criterion,
                                         timeout=timeout, max_trials=max_trials)
+            model: nn.Module = pl_model
             if isinstance(pl_model, LightningModuleFromTorch):
                 # LightningModuleFromTorch.forward fails to trace in FX, so replace it temporarily
-                quantizer.model = pl_model.model
-            else:
-                quantizer.model = pl_model
-
-            if val_dataloader and metric:
-                from bigdl.nano.quantization.quantization_inc import TorchMetricForINC
-                quantizer.eval_dataloader = val_dataloader
-                quantizer.metric = common.Metric(TorchMetricForINC, metric=metric)
-            if approach == 'post_training_static_quant':
-                assert calib_dataloader, "calib_dataloader must not be None when approach is " \
-                                         "post-training static quantization."
-                quantizer.calib_dataloader = calib_dataloader
-            quantized = quantizer()
-            if quantized:
-                return quantized.model
-            else:
-                raise RuntimeError("Found no quantized model satisfying accuracy criterion.")
+                model = pl_model.model
+
+            TorchINCMetric.metric = metric
+            quantizer.post_training_quantize(model, calib_dataloader, val_dataloader,
+                                             TorchINCMetric)
+            return quantizer().model
```

Comment:
why we need to call `quantizer()` again? I think it has been called in `quantizer.post_training_quantize`?